### PR TITLE
[JN-939] Playwright tests addition

### PR DESCRIPTION
**Pull request by aweng98**
#### DESCRIPTION

1. Re-implementation of page interface and abstract classes for different types of pages and components.
2. Turned `constants-en.ts` file to `ourhealth-en.json`.
3. New `modal.ts` component class.

juniper e2e-test uml diagram

### CODE REVIEW

File: e2e-tests/README.md [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 ### Set up
-* Install dependencies and web browsers in **/e2e-tests** dir.
-  >
-  >  * cd e2e-tests/
-  >  * npm install
-  >  * npx playwright install --with-deps
-  >
+* Install dependencies and Playwright web browsers in **/e2e-tests** dir.
+```shell
+npm installn && npx playwright install --with-deps
```
Comment (connorlbark) [inline e2e-tests/README.md:6]: nit: `installn` typo

**Comment by aweng98** [comment]
Hello all, is there any concern with this PR? No review so far.

**Review by connorlbark — APPROVED** [review]
I'm getting a couple failures but they seem to be pretty simple fixes. Here are the ones that are failing:

```
  2 failed
    [OurHealth-Chrome] › src/tests/studies/ourhealth/home-ui.visual.test.ts:7:7 › Home page › UI @visual @ourhealth
    [OurHealth-Chrome] › src/tests/studies/ourhealth/mailing-list.test.ts:4:5 › shows mailing list dialog on direct link
  5 passed (16.5s)
```

That being said, I'm happy to approve for now because we will likely be doing a lot of work in the e2e space soon and you have already done a lot to help us out 😄

**Review by devonbush — APPROVED** [review]
sorry for the slow turnaround Alex.  Just pushed some changes to fix tests that were broken on Connor and my machines due to pixel margins.  Thanks for all this work

**Comment by aweng98** [comment]
> sorry for the slow turnaround Alex. Just pushed some changes to fix tests that were broken on Connor and my machines due to pixel margins. Thanks for all this work

Thank you so much for the fixes. Note about pixel mismatch failures, likely due to use of an external monitor (I generated screenshots on mac laptop). To work aournd this issue, set the viewport size in playwright.config.ts. 
Let me know if I can help with any questions or PR review in future.

**Pull request merged**